frontend/src/store/selectors.ts

```diff
@@ -165,23 +167,89 @@ export const getFocusViewProps: SelectorOf<FocusViewProps> = createSelector(
-      .forEach((task) => {
-        const filteredUncompletedTask = getFilteredNotCompletedInFocusTask(task);
-        const filteredCompletedTask = getFilteredCompletedInFocusTask(task);
-        const { id, order } = task;
-        if (filteredCompletedTask != null && filteredUncompletedTask != null) {
-          taskMetaDataList.push({ id, order, inFocusView: true, inCompleteFocusView: true });
-          taskMetaDataList.push({ id, order, inFocusView: true, inCompleteFocusView: false });
-        } else if (filteredCompletedTask != null) {
-          taskMetaDataList.push({ id, order, inFocusView: true, inCompleteFocusView: true });
-        } else if (filteredUncompletedTask != null) {
-          taskMetaDataList.push({ id, order, inFocusView: true, inCompleteFocusView: false });
+    let oneTime: Task[] = [];
+    const masterTemp: Task[] = [];
+    const group: Task[] = [];
+    const diffGroupType: Record<string, Task[]> = {};
+    const finalSorted: Task[] = [];
+    let final: Map<string, Task<TaskMetadata>> = Map();
+    let test: Map<string, Task<TaskMetadata>> = Map();
+    // const groupMap : Record<  "ONE_TIME" | "MASTER_TEMPLATE" | "GROUP", Task[]>= {};
+    Array.from(tasks.values()).forEach((task) => {
+      const filteredUncompletedTask = getFilteredNotCompletedInFocusTask(task);
+      const filteredCompletedTask = getFilteredCompletedInFocusTask(task);
+      const { id, order } = task;
+      if (filteredCompletedTask != null && filteredUncompletedTask != null) {
+        taskMetaDataList.push({ id, order, inFocusView: true, inCompleteFocusView: true });
+        taskMetaDataList.push({ id, order, inFocusView: true, inCompleteFocusView: false });
+      } else if (filteredCompletedTask != null) {
+        taskMetaDataList.push({ id, order, inFocusView: true, inCompleteFocusView: true });
+      } else if (filteredUncompletedTask != null) {
+        taskMetaDataList.push({ id, order, inFocusView: true, inCompleteFocusView: false });
+      } else {
+        taskMetaDataList.push({ id, order, inFocusView: false, inCompleteFocusView: false });
+      }
+      if (task.metadata.type === 'GROUP') {
+        group.push(task);
+      }
+
+      if (task.metadata.type === 'ONE_TIME') {
+        oneTime.push(task);
+      }
+
+      if (task.metadata.type === 'MASTER_TEMPLATE') {
+        masterTemp.push(task);
+      }
+      final.set(task.id, task);
+    });
+
+    group.forEach((task) => {
+      if (task.metadata.type === 'GROUP') {
+        if (diffGroupType[task.metadata.group]) {
+          diffGroupType[task.metadata.group].push(task);
         } else {
-          taskMetaDataList.push({ id, order, inFocusView: false, inCompleteFocusView: false });
+          diffGroupType[task.metadata.group] = [task];
         }
+      }
+    });
+
+    Object.keys(diffGroupType).sort();
+    // eslint-disable-next-line no-restricted-syntax
+    for (const [key, value] of Object.entries(diffGroupType)) {
```

Comment:
Use Object.entries(diffGroupType).forEach(([key, value]) => { ... }) instead of avoid suppressing linter warnings